src/app/common/components/MustGatherBtn.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from 'react';
+import { Button, Tooltip } from '@patternfly/react-core';
+import { WarningTriangleIcon } from '@patternfly/react-icons';
+import { MustGatherContext } from '@app/common/context';
+import { getMustGatherApiUrl } from '@app/queries/helpers';
+import { IMustGatherResponse } from '@app/client/types';
+import { UseQueryResult } from 'react-query';
+import { IPlan } from '@app/queries/types';
+
+interface IMustGatherBtn {
+  mg: IMustGatherResponse | undefined;
+  mustGathers: UseQueryResult<IMustGatherResponse[], Response>;
+  customName: string;
+  type: 'plan' | 'vm';
+}
+
+export const MustGatherBtn: React.FunctionComponent<IMustGatherBtn> = ({
+  mg,
+  mustGathers,
+  customName,
+  type,
+}) => {
+  const {
+    setMustGatherModalOpen,
+    setActiveMustGather,
+    mustGatherWatchList,
+    setMustGatherWatchList,
+  } = React.useContext(MustGatherContext);
+
+  return (
+    <>
+      {mg?.status === 'completed' && mg?.['archive-name'] ? (
+        <Tooltip
+          content={
+            !mustGathers.isSuccess
+              ? `Could not reach must gather service.`
+              : `${mg?.['archive-name']} available for download`
+          }
+        >
+          <Button
+            isAriaDisabled={!mustGathers.isSuccess}
+            variant="secondary"
+            component={'a'}
+            target="_blank"
+            download
+            href={getMustGatherApiUrl(`must-gather/${mg?.['id']}/data`)}
+          >
+            Download logs
+          </Button>
+        </Tooltip>
+      ) : (
+        <>
+          <Tooltip
+            content={
+              !mustGathers.isSuccess
+                ? `Could not reach must gather service.`
```

Comment:
Just an observation, not a criticism: this is a perfect use case for nested ternaries and would be a mess without them, but SonarCloud is yelling at you to not do that... I really want to figure out how to turn off some of those rules.

Reply:
Agree!